org.elixir_lang.sdk.erlang_dependent.Type.isRootTypeApplicable does not use protected access to JavadocOrderRootType.getInstance()

After an update I was able to configure Internal Erlang SDK, but not Elixir SDK: an error occurs.

Elixir plugin v7.2.0
PhpStorm 2017.3.1
Build #PS-173.3942.32, built on December 12, 2017
JRE: 1.8.0_152-release-1024-b8 x86_64
JVM: OpenJDK 64-Bit Server VM by JetBrains s.r.o
Mac OS X 10.13.2

Elixir v1.5.3 and Erlang v20.2.2, both installed via Homebrew.

Stacktrace:
```
Root type class com.intellij.openapi.roots.JavadocOrderRootType not found. All roots: [Root LIBRARY_FILE]
java.lang.AssertionError: Root type class com.intellij.openapi.roots.JavadocOrderRootType not found. All roots: [Root LIBRARY_FILE]
	at com.intellij.openapi.roots.OrderRootType.getOrderRootType(OrderRootType.java:126)
	at com.intellij.openapi.roots.JavadocOrderRootType.getInstance(JavadocOrderRootType.java:35)
	at org.elixir_lang.sdk.erlang.Type.isRootTypeApplicable(Type.java:148)
	at org.elixir_lang.sdk.elixir.Type.codePathsFromInternalErlangSdk(Type.java:231)
	at org.elixir_lang.sdk.elixir.Type.addNewCodePathsFromInternErlangSdk(Type.java:196)
	at org.elixir_lang.sdk.elixir.Type.configureInternalErlangSdk(Type.java:187)
	at org.elixir_lang.sdk.elixir.Type.configureSdkPaths(Type.java:173)
	at org.elixir_lang.sdk.elixir.Type.setupSdkPaths(Type.java:608)
	at com.intellij.openapi.projectRoots.SdkType.setupSdkPaths(SdkType.java:102)
	at com.intellij.openapi.roots.ui.configuration.projectRoot.ProjectSdksModel.a(ProjectSdksModel.java:290)
	at com.intellij.openapi.roots.ui.configuration.projectRoot.ProjectSdksModel.a(ProjectSdksModel.java:273)
	at com.intellij.openapi.projectRoots.impl.DependentSdkType.a(DependentSdkType.java:87)
	at com.intellij.openapi.projectRoots.impl.SdkConfigurationUtil.a(SdkConfigurationUtil.java:297)
	at com.intellij.ui.mac.MacPathChooserDialog.choose(MacPathChooserDialog.java:174)
	at com.intellij.openapi.fileChooser.FileChooser.chooseFiles(FileChooser.java:114)
	at com.intellij.openapi.fileChooser.FileChooser.chooseFiles(FileChooser.java:91)
	at com.intellij.openapi.projectRoots.impl.SdkConfigurationUtil.selectSdkHome(SdkConfigurationUtil.java:294)
	at com.intellij.openapi.projectRoots.impl.DependentSdkType.createSdkOfType(DependentSdkType.java:82)
	at com.intellij.openapi.projectRoots.impl.DependentSdkType.showCustomCreateUI(DependentSdkType.java:69)
	at com.intellij.openapi.projectRoots.SdkType.showCustomCreateUI(SdkType.java:258)
	at com.intellij.openapi.roots.ui.configuration.projectRoot.ProjectSdksModel.doAdd(ProjectSdksModel.java:273)
	at com.intellij.openapi.roots.ui.configuration.projectRoot.ProjectSdksModel.doAdd(ProjectSdksModel.java:267)
	at org.elixir_lang.facet.sdks.Configurable.addSdk(Configurable.kt:121)
	at org.elixir_lang.facet.sdks.Configurable.access$addSdk(Configurable.kt:22)
	at org.elixir_lang.facet.sdks.Configurable$createComponent$decorator$1.run(Configurable.kt:55)
	at org.elixir_lang.facet.sdks.Configurable$createComponent$decorator$1.run(Configurable.kt:54)
	at com.intellij.ui.ToolbarDecorator$2.doAdd(ToolbarDecorator.java:440)
	at com.intellij.ui.CommonActionsPanel$Buttons.performAction(CommonActionsPanel.java:72)
	at com.intellij.ui.CommonActionsPanel$MyActionButton.actionPerformed(CommonActionsPanel.java:263)
	at com.intellij.openapi.actionSystem.ex.ActionUtil$1.run(ActionUtil.java:220)
	at com.intellij.openapi.actionSystem.ex.ActionUtil.performActionDumbAware(ActionUtil.java:237)
	at com.intellij.openapi.actionSystem.impl.ActionButton.a(ActionButton.java:193)
	at com.intellij.openapi.actionSystem.impl.ActionButton.a(ActionButton.java:151)
	at com.intellij.openapi.actionSystem.impl.ActionButton.processMouseEvent(ActionButton.java:344)
	at java.awt.Component.processEvent(Component.java:6306)
	at java.awt.Container.processEvent(Container.java:2237)
	at java.awt.Component.dispatchEventImpl(Component.java:4897)
	at java.awt.Container.dispatchEventImpl(Container.java:2295)
	at java.awt.Component.dispatchEvent(Component.java:4719)
	at java.awt.LightweightDispatcher.retargetMouseEvent(Container.java:4889)
	at java.awt.LightweightDispatcher.processMouseEvent(Container.java:4526)
	at java.awt.LightweightDispatcher.dispatchEvent(Container.java:4467)
	at java.awt.Container.dispatchEventImpl(Container.java:2281)
	at java.awt.Window.dispatchEventImpl(Window.java:2746)
	at java.awt.Component.dispatchEvent(Component.java:4719)
	at java.awt.EventQueue.dispatchEventImpl(EventQueue.java:764)
	at java.awt.EventQueue.access$500(EventQueue.java:98)
	at java.awt.EventQueue$3.run(EventQueue.java:715)
	at java.awt.EventQueue$3.run(EventQueue.java:709)
	at java.security.AccessController.doPrivileged(Native Method)
	at java.security.ProtectionDomain$JavaSecurityAccessImpl.doIntersectionPrivilege(ProtectionDomain.java:80)
	at java.security.ProtectionDomain$JavaSecurityAccessImpl.doIntersectionPrivilege(ProtectionDomain.java:90)
	at java.awt.EventQueue$4.run(EventQueue.java:737)
	at java.awt.EventQueue$4.run(EventQueue.java:735)
	at java.security.AccessController.doPrivileged(Native Method)
	at java.security.ProtectionDomain$JavaSecurityAccessImpl.doIntersectionPrivilege(ProtectionDomain.java:80)
	at java.awt.EventQueue.dispatchEvent(EventQueue.java:734)
	at com.intellij.ide.IdeEventQueue.d(IdeEventQueue.java:822)
	at com.intellij.ide.IdeEventQueue._dispatchEvent(IdeEventQueue.java:646)
	at com.intellij.ide.IdeEventQueue.dispatchEvent(IdeEventQueue.java:366)
	at java.awt.EventDispatchThread.pumpOneEventForFilters(EventDispatchThread.java:201)
	at java.awt.EventDispatchThread.pumpEventsForFilter(EventDispatchThread.java:116)
	at java.awt.EventDispatchThread.pumpEventsForFilter(EventDispatchThread.java:109)
	at java.awt.WaitDispatchSupport$2.run(WaitDispatchSupport.java:190)
	at java.awt.WaitDispatchSupport$4.run(WaitDispatchSupport.java:235)
	at java.awt.WaitDispatchSupport$4.run(WaitDispatchSupport.java:233)
	at java.security.AccessController.doPrivileged(Native Method)
	at java.awt.WaitDispatchSupport.enter(WaitDispatchSupport.java:233)
	at java.awt.Dialog.show(Dialog.java:1084)
	at com.intellij.openapi.ui.impl.DialogWrapperPeerImpl$MyDialog.show(DialogWrapperPeerImpl.java:734)
	at com.intellij.openapi.ui.impl.DialogWrapperPeerImpl.show(DialogWrapperPeerImpl.java:456)
	at com.intellij.openapi.ui.DialogWrapper.invokeShow(DialogWrapper.java:1696)
	at com.intellij.openapi.ui.DialogWrapper.show(DialogWrapper.java:1645)
	at com.intellij.openapi.options.newEditor.SettingsDialog.a(SettingsDialog.java:83)
	at com.intellij.openapi.application.TransactionGuardImpl.a(TransactionGuardImpl.java:88)
	at com.intellij.openapi.application.TransactionGuardImpl.submitTransactionAndWait(TransactionGuardImpl.java:153)
	at com.intellij.openapi.options.newEditor.SettingsDialog.show(SettingsDialog.java:83)
	at com.intellij.ide.actions.ShowSettingsUtilImpl.showSettingsDialog(ShowSettingsUtilImpl.java:154)
	at org.elixir_lang.inspection.SetupSDKNotificationProvider$Companion.showSmallIDEFacetSettings(SetupSDKNotificationProvider.kt:90)
	at org.elixir_lang.inspection.SetupSDKNotificationProvider$Companion$createSmallIDEFacetPanel$$inlined$apply$lambda$1.run(SetupSDKNotificationProvider.kt:97)
	at com.intellij.ui.EditorNotificationPanel$1.hyperlinkActivated(EditorNotificationPanel.java:126)
	at com.intellij.ui.HyperlinkAdapter.hyperlinkUpdate(HyperlinkAdapter.java:28)
	at com.intellij.ui.HyperlinkLabel.fireHyperlinkEvent(HyperlinkLabel.java:212)
	at com.intellij.ui.HyperlinkLabel.processMouseEvent(HyperlinkLabel.java:136)
	at java.awt.Component.processEvent(Component.java:6306)
	at java.awt.Container.processEvent(Container.java:2237)
	at java.awt.Component.dispatchEventImpl(Component.java:4897)
	at java.awt.Container.dispatchEventImpl(Container.java:2295)
	at java.awt.Component.dispatchEvent(Component.java:4719)
	at java.awt.LightweightDispatcher.retargetMouseEvent(Container.java:4889)
	at java.awt.LightweightDispatcher.processMouseEvent(Container.java:4523)
	at java.awt.LightweightDispatcher.dispatchEvent(Container.java:4467)
	at java.awt.Container.dispatchEventImpl(Container.java:2281)
	at java.awt.Window.dispatchEventImpl(Window.java:2746)
	at java.awt.Component.dispatchEvent(Component.java:4719)
	at java.awt.EventQueue.dispatchEventImpl(EventQueue.java:764)
	at java.awt.EventQueue.access$500(EventQueue.java:98)
	at java.awt.EventQueue$3.run(EventQueue.java:715)
	at java.awt.EventQueue$3.run(EventQueue.java:709)
	at java.security.AccessController.doPrivileged(Native Method)
	at java.security.ProtectionDomain$JavaSecurityAccessImpl.doIntersectionPrivilege(ProtectionDomain.java:80)
	at java.security.ProtectionDomain$JavaSecurityAccessImpl.doIntersectionPrivilege(ProtectionDomain.java:90)
	at java.awt.EventQueue$4.run(EventQueue.java:737)
	at java.awt.EventQueue$4.run(EventQueue.java:735)
	at java.security.AccessController.doPrivileged(Native Method)
	at java.security.ProtectionDomain$JavaSecurityAccessImpl.doIntersectionPrivilege(ProtectionDomain.java:80)
	at java.awt.EventQueue.dispatchEvent(EventQueue.java:734)
	at com.intellij.ide.IdeEventQueue.d(IdeEventQueue.java:822)
	at com.intellij.ide.IdeEventQueue._dispatchEvent(IdeEventQueue.java:646)
	at com.intellij.ide.IdeEventQueue.dispatchEvent(IdeEventQueue.java:366)
	at java.awt.EventDispatchThread.pumpOneEventForFilters(EventDispatchThread.java:201)
	at java.awt.EventDispatchThread.pumpEventsForFilter(EventDispatchThread.java:116)
	at java.awt.EventDispatchThread.pumpEventsForHierarchy(EventDispatchThread.java:105)
	at java.awt.EventDispatchThread.pumpEvents(EventDispatchThread.java:101)
	at java.awt.EventDispatchThread.pumpEvents(EventDispatchThread.java:93)
	at java.awt.EventDispatchThread.run(EventDispatchThread.java:82)
```

--- a/src/org/elixir_lang/sdk/Type.java
+++ b/src/org/elixir_lang/sdk/Type.java
@@ -1,18 +1,30 @@
 package org.elixir_lang.sdk;
 
 import com.intellij.openapi.projectRoots.SdkModificator;
+import com.intellij.openapi.roots.JavadocOrderRootType;
 import com.intellij.openapi.roots.OrderRootType;
 import com.intellij.openapi.vfs.LocalFileSystem;
 import com.intellij.openapi.vfs.VirtualFile;
 import org.jetbrains.annotations.NotNull;
+import org.jetbrains.annotations.Nullable;
 
 import java.nio.file.Path;
 import java.util.function.Consumer;
 
 import static org.elixir_lang.sdk.HomePath.eachEbinPath;
+import static org.elixir_lang.sdk.ProcessOutput.isSmallIde;
 
 public class Type {
     private Type() {
+    }
+
+    public static void addCodePaths(@NotNull SdkModificator sdkModificator) {
+        eachEbinPath(
+                sdkModificator.getHomePath(),
+                ebin -> ebinPathChainVirtualFile(
+                        ebin, virtualFile -> sdkModificator.addRoot(virtualFile, OrderRootType.CLASSES)
+                )
+        );
     }
 
     public static void ebinPathChainVirtualFile(@NotNull Path ebinPath, Consumer<VirtualFile> virtualFileConsumer) {
@@ -25,12 +37,16 @@ public class Type {
         }
     }
 
-    public static void addCodePaths(@NotNull SdkModificator sdkModificator) {
-        eachEbinPath(
-                sdkModificator.getHomePath(),
-                ebin -> ebinPathChainVirtualFile(
-                        ebin, virtualFile -> sdkModificator.addRoot(virtualFile, OrderRootType.CLASSES)
-                )
-        );
+    @Nullable
+    public static OrderRootType documentationRootType() {
+        OrderRootType rootType;
+
+        if (isSmallIde()) {
+            rootType = null;
+        } else {
+            rootType = JavadocOrderRootType.getInstance();
+        }
+
+        return rootType;
     }
 }

--- a/src/org/elixir_lang/sdk/elixir/Type.java
+++ b/src/org/elixir_lang/sdk/elixir/Type.java
@@ -52,6 +52,7 @@ import static org.elixir_lang.sdk.ProcessOutput.STANDARD_TIMEOUT;
 import static org.elixir_lang.sdk.ProcessOutput.isSmallIde;
 import static org.elixir_lang.sdk.ProcessOutput.transformStdoutLine;
 import static org.elixir_lang.sdk.Type.addCodePaths;
+import static org.elixir_lang.sdk.Type.documentationRootType;
 import static org.elixir_lang.sdk.Type.ebinPathChainVirtualFile;
 
 public class Type extends org.elixir_lang.sdk.erlang_dependent.Type {
@@ -108,19 +109,6 @@ public class Type extends org.elixir_lang.sdk.erlang_dependent.Type {
                 sdkModificator.addRoot(hexdocUrlVirtualFile, documentationRootType);
             }
         }
-    }
-
-    @Nullable
-    public static OrderRootType documentationRootType() {
-        OrderRootType rootType;
-
-        if (isSmallIde()) {
-            rootType = null;
-        } else {
-            rootType = JavadocOrderRootType.getInstance();
-        }
-
-        return rootType;
     }
 
     private static void addDocumentationPath(@NotNull SdkModificator sdkModificator,

--- a/src/org/elixir_lang/sdk/elixir/conversion/Converter.kt
+++ b/src/org/elixir_lang/sdk/elixir/conversion/Converter.kt
@@ -8,6 +8,7 @@ import com.intellij.openapi.projectRoots.Sdk
 import com.intellij.openapi.projectRoots.SdkModificator
 import com.intellij.openapi.roots.OrderRootType
 import com.intellij.openapi.vfs.VirtualFileManager
+import org.elixir_lang.sdk.Type.documentationRootType
 import org.elixir_lang.sdk.elixir.Type
 import org.elixir_lang.sdk.elixir.Type.*
 import org.elixir_lang.sdk.erlang_dependent.SdkAdditionalData

--- a/src/org/elixir_lang/sdk/erlang/Type.java
+++ b/src/org/elixir_lang/sdk/erlang/Type.java
@@ -7,7 +7,6 @@ import com.intellij.openapi.projectRoots.Sdk;
 import com.intellij.openapi.projectRoots.SdkModel;
 import com.intellij.openapi.projectRoots.SdkModificator;
 import com.intellij.openapi.projectRoots.SdkType;
-import com.intellij.openapi.roots.JavadocOrderRootType;
 import com.intellij.openapi.roots.OrderRootType;
 import com.intellij.openapi.util.SystemInfo;
 import com.intellij.openapi.util.Version;
@@ -26,6 +25,7 @@ import java.util.regex.Pattern;
 
 import static org.elixir_lang.sdk.HomePath.*;
 import static org.elixir_lang.sdk.Type.addCodePaths;
+import static org.elixir_lang.sdk.Type.documentationRootType;
 
 /**
  * An Erlang SdkType for use when `intellij-erlang` is not installed
@@ -145,7 +145,7 @@ public class Type extends SdkType {
     public boolean isRootTypeApplicable(@NotNull OrderRootType type) {
         return type == OrderRootType.CLASSES ||
                 type == OrderRootType.SOURCES ||
-                type == JavadocOrderRootType.getInstance();
+                type == documentationRootType();
     }
 
     @Override

